code/modules/projectiles/guns/rifles.dm

```diff
@@ -933,6 +933,117 @@
+	)
+
+/obj/item/weapon/gun/rifle/m16/grenadier/handle_starting_attachment()
+	..()
+	var/obj/item/attachable/attached_gun/grenade/m203/integrated = new(src)
+	integrated.flags_attach_features &= ~ATTACH_REMOVABLE
+	integrated.hidden = FALSE
+	integrated.Attach(src)
+	update_attachable(integrated.slot)
+
+//-------------------------------------------------------
+//XM177 carbine
+//awesome vietnam era special forces carbine version of the M16
+
+/obj/item/weapon/gun/rifle/xm177
+	name = "\improper XM177E2 carbine"
+	desc = "An old design, essentially a shortened M16A1 with a collapsable stock. It is chambered in 5.56x45mm. The short length inhibits the attachment of most underbarrel attachments, and the barrel moderator prohibits the attachment of all muzzle devices."
+	desc_lore = "A carbine similar to the M16A1, with a collapsible stock and a distinct flash suppressor. A stamp on the receiver reads: 'COLT AR-15 - PROPERTY OF U.S. GOVT - XM177E2 - CAL 5.56MM' \nA design originating from the Vietnam War, the XM177, also known as the Colt Commando or GAU-5/A, was an improvement on the CAR-15 Model 607, fixing multiple issues found with the limited service of the Model 607 with Special Forces. The XM177 saw primary use with Army Special Forces and Navy Seals operating as commandos. \nHow this got here is a mystery."
+	icon = 'icons/obj/items/weapons/guns/guns_by_faction/colony.dmi'
+	icon_state = "xm177"
+	item_state = "m16"
+	current_mag = /obj/item/ammo_magazine/rifle/m16
+
+	flags_gun_features = GUN_CAN_POINTBLANK|GUN_ANTIQUE
+
+	fire_sound = 'sound/weapons/gun_m16.ogg'
+	reload_sound = 'sound/weapons/handling/gun_m16_reload.ogg'
+	unload_sound = 'sound/weapons/handling/gun_m16_unload.ogg'
+	accepted_ammo = list(
+		/obj/item/ammo_magazine/rifle/m16,
+		/obj/item/ammo_magazine/rifle/m16/ap,
+	)
+
+	attachable_allowed = list(
+		/obj/item/attachable/reddot,
+		/obj/item/attachable/reflex,
+		/obj/item/attachable/scope/mini,
+		/obj/item/attachable/verticalgrip,
+		/obj/item/attachable/flashlight,
+		/obj/item/attachable/lasersight,
+		/obj/item/attachable/stock/m16/xm177,
+	)
+
+	random_spawn_chance = 75
+	random_spawn_rail = list(
+		/obj/item/attachable/reddot,
+		/obj/item/attachable/reflex/,
+		/obj/item/attachable/flashlight,
+	)
+	random_spawn_under = list(
+		/obj/item/attachable/verticalgrip,
+		/obj/item/attachable/lasersight,
+	)
+
+/obj/item/weapon/gun/rifle/xm177/set_gun_attachment_offsets()
+	attachable_offset = list("muzzle_x" = 30, "muzzle_y" = 18,"rail_x" = 9, "rail_y" = 20, "under_x" = 19, "under_y" = 13, "stock_x" = 15, "stock_y" = 14)
+
+/obj/item/weapon/gun/rifle/xm177/handle_starting_attachment()
+	..()
+	var/obj/item/attachable/stock/m16/xm177/integrated = new(src)
+	integrated.hidden = FALSE
+	integrated.flags_attach_features &= ~ATTACH_REMOVABLE
+	integrated.Attach(src)
+	update_attachable(integrated.slot)
+
+/obj/item/weapon/gun/rifle/xm177/set_gun_config_values()
+	..()
+	fire_delay = FIRE_DELAY_TIER_SMG
```

Comment:
Could you give a good reasoning as to the stats and why the gun should feel this way? Tldr. justify this.

Reply:
Awaiting testing by nanu, per discord.